src/http/request_context.rs

```diff
@@ -93,4 +129,18 @@ mod test {
     req_ctx.set_min_max_age(120);
     assert_eq!(req_ctx.get_min_max_age(), Some(120));
   }
+
+  #[test]
+  fn test_update_cache_visibility_private() {
+    let req_ctx = RequestContext::default();
+    req_ctx.set_cache_visibility(&Some(Cachability::Private));
+    assert_eq!(req_ctx.is_cache_public(), Some(false));
+  }
+
+  #[test]
+  fn test_update_cache_visibility_public() {
+    let req_ctx = RequestContext::default();
+    req_ctx.set_cache_visibility(&Some(Cachability::Public));
+    assert_eq!(req_ctx.is_cache_public(), None);
```

Comment:
Hi @ologbonowiwi , by default, the cache is assumed to be `public`. I wanted to avoid locking for updating the default field, so considered both `Some(true)` and `None` as `public`. Would that be a fair tradeoff to avoid unnecessary locking or do you think it's better to keep it restricted to just `Some(true)`?

Reply:
imo we should have three statuses:

- public: `Some(true)`
- private: `Some(false)`
- no cache: `None`

but it'd be nice to hear @tusharmath and/or @amitksingh1490 opinion on this one

Reply:
@A-N-uraag @ologbonowiwi , we could keep public as default to avoid locking.

Reply:
do we even have the situation when we set it to Some(true)? if no, it could become just a boolean flag. And in case you'll need 3 states consider using enum instead of Option(bool)

Reply:
Hi @meskill, I tried to avoid locking at all in the first place and use a simple bool(since there's no "race condition" here technically). I am not very familiar with `rust` and this was resulting in a mutability issue which meant changing a lot of functions to use mutability. So, I ended up using something similar to `max_age`. If you have suggestions on improving this, I would love to work on them. Thanks.

Reply:
that the mutex that provides internal mutability for you, not Option. So, when you have mutex you can change the value, and Option in that case in just excessive wrapper for value